zerver/lib/events.py

```diff
@@ -1350,12 +1378,19 @@ def _draft_update_action(i: int) -> None:
                     "can_create_private_channel_group",
                     "can_create_web_public_channel_group",
                 ]:
+                    if system_groups_name_dict is None:
+                        system_groups = get_role_based_system_groups_dict(user_profile.realm)
```

Comment:
This adds a DB query in `apply_event` code, which I think we try to avoid, but should be rare case.

Reply:
That's probably OK, since it's a rare situation. 

Just add a comment saying 

`# Here we do a database query, because `get_corresponding_policy_value_for_group_setting` requires the full set of system groups. This could be avoided if `realm_user_group` were in `fetch_event_types`, since the system groups should all be there, but the query itself is cheap enough that it's likely not worth that complexity.